scriptworker/cot/verify.py

```diff
@@ -1145,7 +1145,7 @@ def verify_link_in_task_graph(chain, decision_link, task_link):
     else:
         tmpl = await get_in_tree_template(decision_link)
         for k in ('action', 'push', 'repository'):
-            jsone_context[k] = deepcopy(action_defn['hookPayload']['decision'][k])
+            jsone_context[k] = jsone.render(action_defn['hookPayload']['decision'][k], jsone_context)
```

Comment:
Looking at the [spec](https://docs.taskcluster.net/docs/manual/using/actions/spec) and the [tools implementation](https://github.com/taskcluster/taskcluster-tools/blob/2d2897bd5e8cab418fbfd518dbcc9c892abfa066/src/views/UnifiedInspector/ActionsMenu.jsx#L480-L502), I think this is incorrect. My reading is that the returned context should *just* be `hookPayload` rendered with the previous context.

Having written that, looking at the hook definition in [ci-admin](https://hg.mozilla.org/build/ci-admin/file/edad9f8f78d42bc547c1f684b0f9dc116ccbd00a/ciadmin/generate/in_tree_actions.py#l155), it looks like the context under goes some more modification before being passed to the in-tree `.taskcluster.yml`.

Reply:
Yeah, I'm not sure what this line is about.  But rendering some sub-property is usually a red flag that you're doing something wrong.

Reply:
Ok. The easy workaround is changing the relpro action symbol to a static string, but it's been very useful to have it be templatized. I'm probably going to ask more questions to figure out how action-hooks are able to double-render this payload on the ci-admin / hook / taskgraph side; once we figure that out, hopefully it'll be easier to replicate on the scriptworker side.

Reply:
There may be a bug - a repro recipe would be helpful!

Reply:
I've pushed a new change that renders the entirety of `hookPayload` instead of subsections. Does this match what happens in taskcluster? Or is this a red flag still?